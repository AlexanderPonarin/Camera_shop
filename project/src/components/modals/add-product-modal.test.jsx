import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import AddProductModal from './add-product-modal';
import { Provider } from 'react-redux';
import {configureMockStore} from '@jedmao/redux-mock-store';
import HistoryRouter from '../history-route/history-route';
import {createMemoryHistory} from 'history';



const mockStore = configureMockStore();

const store = mockStore({
  DATA: {
    isQuestionsDataLoading: false,
  },
  MODALVIEW: {
    addItemModalViewStatus: false
  }
});
describe('AddProductModal', () => {
  it('should render with correct product information', () => {
    const mockProduct = {
      name: 'Test Product',
      previewImg: 'test.png',
      previewImg2x: 'test@2x.png',
      previewImgWebp: 'test.webp',
      previewImgWebp2x: 'test@2x.webp',
      vendorCode: '123456',
      type: 'Test Type',
      level: 'Test Level',
      price: '99.99',
    };

    render(
      <Provider>
        <AddProductModal product={mockProduct} />
      </Provider>
    );

    expect(screen.getByText(/Добавить товар в корзину/i)).toBeInTheDocument();
    expect(screen.getByText(mockProduct.name)).toBeInTheDocument();
    expect(screen.getByText(`Артикул: ${mockProduct.vendorCode}`)).toBeInTheDocument();
    expect(screen.getByText(mockProduct.type)).toBeInTheDocument();
    expect(screen.getByText(mockProduct.level)).toBeInTheDocument();
    expect(screen.getByText(`Цена: ${mockProduct.price}`)).toBeInTheDocument();
    expect(screen.getByAltText(mockProduct.name)).toHaveAttribute('src', `/${mockProduct.previewImg}`);
    expect(screen.getByLabelText(/закрыть попап/i)).toBeInTheDocument();
  });
});
