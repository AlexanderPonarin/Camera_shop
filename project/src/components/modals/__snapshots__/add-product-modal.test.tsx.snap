// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProductModal should match snapshot 1`] = `
<div
  class="modal"
  data-testid="add-product-modal"
>
  <div
    class="modal__wrapper"
  >
    <div
      class="modal__overlay"
    />
    <div
      class="modal__content"
    >
      <p
        class="title title--h4"
      >
        Добавить товар в корзину
      </p>
      <div
        class="basket-item basket-item--short"
      >
        <div
          class="basket-item__img"
        >
          <picture>
            <source
              srcset="/testImg.webp, /testImg@2x.webp"
              type="image/webp"
            />
            <img
              alt="Test product"
              height="120"
              src="/testImg.jpg"
              srcset="/testImg@2x.jpg"
              width="140"
            />
          </picture>
        </div>
        <div
          class="basket-item__description"
        >
          <p
            class="basket-item__title"
          >
            Test product
          </p>
          <ul
            class="basket-item__list"
          >
            <li
              class="basket-item__list-item"
            >
              <span
                class="basket-item__article"
              >
                Артикул:
              </span>
              <span
                class="basket-item__number"
              >
                12345
              </span>
            </li>
            <li
              class="basket-item__list-item"
            >
              test type
            </li>
            <li
              class="basket-item__list-item"
            >
              beginner
            </li>
          </ul>
          <p
            class="basket-item__price"
          >
            <span
              class="visually-hidden"
            >
              Цена:
            </span>
            10
          </p>
        </div>
      </div>
      <div
        class="modal__buttons"
      >
        <button
          class="btn btn--purple modal__btn modal__btn--fit-width"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            height="16"
            width="24"
          >
            <use
              xlink:href="#icon-add-basket"
            />
          </svg>
          Добавить в корзину
        </button>
      </div>
      <button
        aria-label="Закрыть попап"
        class="cross-btn"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          height="10"
          width="10"
        >
          <use
            xlink:href="#icon-close"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
